xmodule/capa/capa_problem.py

```diff
@@ -481,16 +495,17 @@ def get_grade_from_current_answers(self, student_answers):
             # student_answers contains a proper answer or the filename of
             # an earlier submission, so for now skip these entirely.
             # TODO: figure out where to get file submissions when rescoring.
-            if 'filesubmission' in responder.allowed_inputfields and student_answers is None:
-                _ = self.capa_system.i18n.gettext
-                raise Exception(_("Cannot rescore problems with possible file submissions"))
-
-            # use 'student_answers' only if it is provided, and if it might contain a file
-            # submission that would not exist in the persisted "student_answers".
-            if 'filesubmission' in responder.allowed_inputfields and student_answers is not None:
-                results = responder.evaluate_answers(student_answers, oldcmap)
+            if 'filesubmission' in responder.allowed_inputfields:
+                # use 'student_answers' only if it is provided, and if it might contain a file
+                # submission that would not exist in the persisted "student_answers".
+                if student_answers is None:
+                    _ = self.capa_system.i18n.gettext
+                    raise Exception(_("Cannot rescore problems with possible file submissions"))
+                results = responder.evaluate_answers(student_answers, correct_map)
+            elif student_answers is not None:
+                results = responder.evaluate_answers(student_answers, correct_map)
```

Comment:
why are we adding this new conditional?

Reply:
As mentioned in [this comment](https://github.com/openedx/edx-platform/pull/33911#discussion_r1498013355), having to do the rescore for each of the student's attempts, we must be able to use the `student_answers` argument, and not always from the instance (`self.student_answers`).

Reply:
I also recommend doing away with this extra conditional through the arguments used by `get_grade_from_current_answers()` when delegating to `get_grade_from_answers()`